lib/Model/Subject/Subject.php

```diff
@@ -140,6 +143,10 @@ public function __get($name)
     public function & __set($name, $value)
     {
         switch ($name) {
+            case 'id':
+                $value = $value ? $value : null;
+                break;
```

Comment:
This should be removed. The `$id` property should only be set by class methods and not modified by external process.